mk-poetry-dep.nix

```diff
@@ -110,6 +110,63 @@ pythonPackages.callPackage
       format = if isWheelUrl then "wheel" else if isDirectory || isGit || isUrl then "pyproject" else fileInfo.format;
 
       hooks = python.pkgs.callPackage ./hooks { };
+
+      src =
+        let
+          srcRoot =
+            if isGit then
+              (
+                builtins.fetchGit ({
+                  inherit (source) url;
+                  rev = source.resolved_reference or source.reference;
+                  ref = sourceSpec.branch or (if sourceSpec ? tag then "refs/tags/${sourceSpec.tag}" else "HEAD");
+                } // (
+                  lib.optionalAttrs
+                    (((sourceSpec ? rev) || (sourceSpec ? branch) || (source ? resolved_reference) || (source ? reference))
+                      && (lib.versionAtLeast builtins.nixVersion "2.4"))
+                    {
+                      allRefs = true;
+                    }) // (
+                  lib.optionalAttrs (lib.versionAtLeast builtins.nixVersion "2.4") {
+                    submodules = true;
+                  })
+                )
+              )
+            else if isWheelUrl then
+              builtins.fetchurl
+                {
+                  inherit (source) url;
+                  sha256 = fileInfo.hash;
+                }
+            else if isUrl then
+              builtins.fetchurl
+                {
+                  inherit (source) url;
+                  sha256 = fileInfo.hash;
+                }
+            else if isDirectory then
+              (poetryLib.cleanPythonSources { src = localDepPath; })
+            else if isFile then
+              localDepPath
+            else if isLegacy then
+              pyproject-nix.fetchers.fetchFromLegacy
+                {
+                  pname = name;
+                  inherit (fileInfo) file hash;
+                  inherit (source) url;
+                }
+            else
+              pyproject-nix.fetchers.fetchFromPypi {
+                pname = name;
+                inherit (fileInfo) file hash;
+                inherit version;
+              };
+        in
+        if source ? subdirectory then
+          srcRoot + "/${source.subdirectory}"
+        else
+          srcRoot;
+        missing_pypproject_toml = ! builtins.pathExists "${src}/pyproject.toml";
```

Comment:
This is IFD (import-from-derivation), and not permitted in restricted eval.
This would break poetry2nix for such setups (e.g. users of Hydra).

As well as just being terrible for eval performance:
https://fzakaria.com/2020/10/20/nix-parallelism-import-from-derivation.html